File: backend/internal/service/project_domain_types.go
package service

import "time"

const (
	ProjectDomainKindManaged = "managed"

	ProjectDomainStatusPending = "pending"
	ProjectDomainStatusActive  = "active"
	ProjectDomainStatusError   = "error"
)

type ProjectDomainRecord struct {
	ID                 string
	ProjectID          string
	Hostname           string
	Label              string
	Kind               string
	Status             string
	StatusReason       string
	CloudflareRecordID string
	TargetKind         string
	TargetID           string
	LastSyncedIP       string
	PublicURL          string
	CreatedAt          time.Time
	UpdatedAt          time.Time
}

type AllocateProjectDomainCommand struct {
	RequesterUserID string
	RequesterRole   string
	ProjectID       string
	Regenerate      bool
}

type RenameProjectDomainCommand struct {
	RequesterUserID string
	RequesterRole   string
	ProjectID       string
	Label           string
}

File: frontend/src/modules/project-domain/project-domain-types.ts
export type ProjectDomain = {
  id: string;
  project_id: string;
  hostname: string;
  label: string;
  kind: string;
  status: 'pending' | 'active' | 'error' | string;
  status_reason?: string;
  cloudflare_record_id?: string;
  target_kind?: string;
  target_id?: string;
  last_synced_ip?: string;
  public_url?: string;
  created_at: string;
  updated_at: string;
};

export type AllocateProjectDomainRequest = {
  regenerate?: boolean;
};

export type RenameProjectDomainRequest = {
  label: string;
};
